fastly/purge.go

```diff
@@ -29,6 +29,7 @@ func (c *Client) Purge(i *PurgeInput) (*Purge, error) {
 	ro := new(RequestOptions)
 	ro.Parallel = true
 	if i.Soft {
+		ro.Headers = make(map[string]string)
 		ro.Headers["Fastly-Soft-Purge"] = "1"
 	}
```

Comment:
Small nit: this doesn't follow the conventions throughout the rets of the project or go idioms. Personally i'd write it as:

```suggestion
	ro := &RequestOptions{}
	ro.Parallel = true
	if i.Soft {
		ro.Headers = map[string]string{
			"Fastly-Soft-Purge": 1
		}
	}
```

Reply:
@phamann happy to accept this suggestion, but before I do, regarding your comment...

> or go idioms

I've just read back over the official "Effective Go" and the only distinction they make to using `new` vs manually constructing a struct is when [you have lots of fields to set as part of the initialisation process](https://golang.org/doc/effective_go#composite_literals). Is there a reason why we've opted for `&RequestOptions{}` instead for this simple case (as we don't set any fields)?

Reply:
Yes, apologies I missed the `Parallel: true`, definitely prefer you suggestion 👍🏻